r55/src/exec.rs

```diff
@@ -171,15 +174,30 @@ fn execute_riscv(
 
                 let Ok(syscall) = Syscall::try_from(t0 as u8) else {
                     println!("Unhandled syscall: {:?}", t0);
-                    return return_revert(interpreter);
+                    return return_revert(interpreter, evm_gas);
                 };
 
                 match syscall {
                     Syscall::Return => {
                         let ret_offset: u64 = emu.cpu.xregs.read(10);
                         let ret_size: u64 = emu.cpu.xregs.read(11);
+
+                        let r55_gas = r55_gas_used(&emu.cpu.inst_counter.clone());
```

Comment:
```suggestion
                        let r55_gas = r55_gas_used(&emu.cpu.inst_counter);
```
Seems like unnecessary clone here.

Reply:
thanks!